datepicker/operations/index.js

```diff
@@ -0,0 +1,205 @@
+var moment = require('moment/min/moment-with-locales.min');
+var Datepicker = require('./../index');
+
+Datepicker.prototype._showView = function(view) {
+  if (this.model.get('activeView') === view) return;
+
+  this.model.set('activeView', view);
+};
+
+Datepicker.prototype._selectDate = function(momentDate, closeOnSelect) {
+  if (this._isOutOfRange(momentDate)) return;
+
+  var selectedDate = this.model.get('selectedDate');
+
+  if (!momentDate.isSame(selectedDate, 'day')) {
+    this._animateText('date', selectedDate);
+    if (!momentDate.isSame(selectedDate, 'year')) this._animateText('year', selectedDate);
+  }
+
+  this.model.set('selectedDate', momentDate);
+  this.emit('selected', this.model.get('selectedDate').format());
+
+  if (closeOnSelect) {
+    this.setValue();
+    this._hide();
+  }
+};
+
+Datepicker.prototype.action = function(action) {
+  this.emit('action', action.text || action);
+
+  if (action.disableDismiss) return;
+
+  this._hide();
+};
+
+Datepicker.prototype._selectYear = function(year) {
+  var momentDate = this.model.get('selectedDate').clone();
+  momentDate.year(year);
+
+  this._selectDate(momentDate);
+  this._setCurrentDate(momentDate);
+  
+  // REVIEW: center on selection or only on view load?
```

Comment:
TBH I have no idea if it's unnecessary to center on selection as well. In general, I think it's good the current date is always the starting point. If the reason behind centering is to "fix it" so that it's not totally off, I think it's safer to do it on selection as well. It will very most likely not be noticeable performance wise anyway so why not be safe then?